src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -5779,25 +5779,26 @@ Copyright (C) Microsoft Corporation. All rights reserved.
       -->
       <TrackFileWritesShareableOutsideOfProjectDirectory Condition=" '$(TrackFileWritesShareableOutsideOfProjectDirectory)' == '' ">false</TrackFileWritesShareableOutsideOfProjectDirectory>
     </PropertyGroup>
-
-    <ItemGroup>
-      <!-- Always track the FileWrites from the project-specific output directories -->
-      <FilesToTrackFromOutputDirectories Include="@(FileWrites)" />
-      <!-- Track the FileWritesShareable too if requested -->
-      <FilesToTrackFromOutputDirectories Include="@(FileWritesShareable)" Condition=" $(TrackFileWritesShareableOutsideOfProjectDirectory) == true "/>
-    </ItemGroup>
     
+    <!-- Even if we don't allow cleaning filewrites from outside of the project bubble, we should still include FileWritesShareable that are inside the bubble -->
     <FindUnderPath Condition="!$(TrackFileWritesShareableOutsideOfProjectDirectory)" Path="$(MSBuildProjectDirectory)" Files="@(FileWritesShareable)" UpdateToAbsolutePaths="true">
       <Output TaskParameter="InPath" ItemName="FileWrites"/>
```

Comment:
This line is the main thing I want to make sure still works - if we mess this up then items from other projects won't get cleaned out of the current project. Essentially the same problem as in https://github.com/dotnet/msbuild/pull/12233.